File: src/doc_analyse/verifier.py
from __future__ import annotations

from collections.abc import Mapping
from dataclasses import dataclass
from typing import Any, Optional

from doc_analyse.classifiers.base import BaseClassifier, ClassificationResult


@dataclass
class DocumentVerifier:
    classifier: BaseClassifier

    def verify_text(
        self,
        text: str,
        metadata: Optional[Mapping[str, Any]] = None,
    ) -> ClassificationResult:
        return self.classifier.classify(text=text, metadata=metadata)

File: src/orchestration/verifier.ts
/**
 * Document verifier — simple wrapper around a classifier.
 * Ported from doc_analyse/verifier.py
 */

import { BaseClassifier, ClassificationResult } from "../classifiers/base.js";

export class DocumentVerifier {
  private readonly classifier: BaseClassifier;

  constructor(classifier: BaseClassifier) {
    this.classifier = classifier;
  }

  verify_text(
    text: string,
    metadata: Record<string, unknown> | null = null,
  ): ClassificationResult {
    return this.classifier.classify(text, metadata);
  }
}
